packages/web-components/src/components/ic-chip/ic-chip.tsx

```diff
@@ -127,6 +139,31 @@ export class Chip {
     this.isHovered = false;
   };
 
+  private setChipColour = () => {
+    let colorRGBA = null;
```

Comment:
This bit of code (143-150) is repeated in both ic-badge and ic-theme. Suggest making it a helper function that can be used in all three.